# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/dns/GetAddrInfo.h
# Commit: af0f14eed70e
# Full Hash: af0f14eed70edb2abb1ecd3cae4928387287d3e1
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2024-06-14 20:49:45
# Regressor Bug: 1901990
# File Overlap Count: 2
# Description:
#   Bug 1901990 - Introduce a pref for setting a mock HTTPS RR, r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D213559
# ==============================================================================

diff -r 9f913cbd60ee -r af0f14eed70e netwerk/dns/GetAddrInfo.h
--- a/netwerk/dns/GetAddrInfo.h	Thu Jun 13 15:08:04 2024 +0200
+++ b/netwerk/dns/GetAddrInfo.h	Fri Jun 14 09:11:49 2024 +0000
@@ -85,6 +85,12 @@
 nsresult ParseHTTPSRecord(nsCString& aHost, DNSPacket& aDNSPacket,
                           TypeRecordResultType& aResult, uint32_t& aTTL);
 
+// Use the provided aHost to create a mock HTTPS record.
+nsresult CreateAndResolveMockHTTPSRecord(const nsACString& aHost,
+                                         uint16_t aFlags,
+                                         TypeRecordResultType& aResult,
+                                         uint32_t& aTTL);
+
 class NativeDNSResolverOverride : public nsINativeDNSResolverOverride {
   NS_DECL_THREADSAFE_ISUPPORTS
   NS_DECL_NSINATIVEDNSRESOLVEROVERRIDE